lib/lists.nix

```diff
@@ -226,6 +226,31 @@ rec {
   */
   all = builtins.all or (pred: foldr (x: y: if pred x then y else false) true);
 
+  #lexicalLessThan = a: b:
+  #  let
+  #    alen = lib.length a;
+  #    compareElements = index:
+  #      if index == alen then false
+  #      else if lib.elemAt a index == lib.elemAt b index then compareElements (index + 1)
+  #      else lib.elemAt a index < lib.elemAt b index;
+  #  in
+  #  if alen == lib.length b then compareElements 0
+  #  else alen < lib.length b;
+
+  lexicalLessThan =
```

Comment:
See also https://github.com/NixOS/nix/pull/5631, which makes lists comparable in Nix directly